src/java/com/google/cloud/bigquery/dwhassessment/extractiontool/subcommand/ExtractSubcommand.java

```diff
@@ -408,6 +389,64 @@ private ExtractExecutor.Arguments getValidatedArguments() {
     return arguments;
   }
 
+  private void setOutputPath() {
+    Path path = Paths.get(outputPathString);
+    if (path.toString().endsWith(".zip")) {
+      if (!Files.isDirectory(path.getParent())) {
+        throw new ParameterException(
+            spec.commandLine(),
+            String.format("Parent path of --output '%s' is not a directory.", path.getParent()));
+      }
+    } else {
+      if (!Files.isDirectory(path)) {
+        throw new ParameterException(
+            spec.commandLine(),
+            String.format("--output must specify a directory, but '%s' is not a directory.", path));
+      }
+    }
+    argumentsBuilder.setOutputPath(path);
+  }
+
+  private void validateAndSetPrevRunPathIncrementalMode() {
+    if (chunkRows < 1) {
+      throw new ParameterException(
+          spec.commandLine(),
+          "Non-normal run modes require chunked processing. Set --rows-per-chunk to a positive"
+              + " integer to enable chunked processing.");
+    }
+    if (prevRunPathString == null) {
+      throw new ParameterException(
+          spec.commandLine(), "--run-mode is not NORMAL but --prev-run-path is unspecified.");
+    }
+    if (prevRunPathString.endsWith(".zip")) {
+      throw new ParameterException(
+          spec.commandLine(), "Incremental mode is not supported for zipped records, yet.");
+    }
+    Path path = Paths.get(prevRunPathString);
+    if (!Files.isDirectory(path)) {
+      throw new ParameterException(
+          spec.commandLine(),
+          String.format(
+              "--prev-run-path must specify a directory, but '%s' is not a directory.", path));
+    }
+    argumentsBuilder.setPrevRunPath(path);
+  }
+
+  private void validateAndSetPrevRunPathRecoveryMode() {
+    String pathString = prevRunPathString == null ? outputPathString : prevRunPathString;
```

Comment:
Could you add a test to cover a case when previous path is specified?

Reply:
Done.